crates/relayer/src/event/monitor.rs

```diff
@@ -85,8 +107,8 @@ pub struct EventMonitor {
     client: WebSocketClient,
     /// Async task handle for the WebSocket client's driver
     driver_handle: JoinHandle<()>,
-    /// Channel to handler where the monitor for this chain sends the events
-    tx_batch: channel::Sender<Result<EventBatch>>,
+    /// Event bus for broadcasting events
+    event_bus: EventBus<Arc<Result<EventBatch>>>,
     /// Channel where to receive client driver errors
```

Comment:
I hope this type of strong tendemint-related type is made more generic so that I don't have to copy a copy to make changes

Reply:
And here's the Query, I think here and abci forced binding

Reply:
You don't have to use the `EventMonitor` as long as you are able to provide a `fn subscribe(&self) -> Result<Subscription, Error>` method in your `ChainEndpoint` implementation, where `Subscription` is a type alias which expands to:

```rust
type Subscription = crossbeam_channel::Receiver<Arc<MonitorResult<EventBatch>>>;
type MonitorResult<T> = Result<T, ibc_relayer::event::monitor::Error>;

pub struct EventBatch {
    pub chain_id: ChainId,
    pub tracking_id: TrackingId,
    pub height: Height,
    pub events: Vec<IbcEventWithHeight>,
}

pub struct IbcEventWithHeight {
    pub event: IbcEvent,
    pub height: Height,
}
```

where `IbcEvent` is a generic IBC event, not specific to Cosmos-based chains.

Would that work for you?

Reply:
look good